### PR TITLE
Updating the release process doc with latest prowjob approach.

### PR DESCRIPTION
**Pull request by vijtrip2**
Issue #, if available:

Description of changes:
Updating the release process doc with latest prowjob approach.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/contents/dev-docs/release.md [hunk 1]
```diff
@@ -6,10 +6,17 @@ Remember that there is no single ACK binary. Rather, when we build a release
...
 binaries, each of which are installed separately.
 
-This documentation covers the steps involved in building a service controller's
-release artifacts, including the Helm charts and binary Docker images for an
-ACK service controller, publishing those artifacts and tagging the Git source
-repository appropriately to create an official "release".
+This documentation covers the steps involved for officially publishing
+a ACK service controller's release artifacts.
+
+Once ACK service controller changes are tested by service team and they wish to
```
Comment (RedbackThomson) [inline docs/contents/dev-docs/release.md:12]: nit: `by the service team`

File: docs/contents/dev-docs/release.md [hunk 2]
```diff
@@ -6,10 +6,17 @@ Remember that there is no single ACK binary. Rather, when we build a release
...
+release latest artifacts, service team only needs to tag the service-controller
+github repository with a semver tag (Ex: v0.0.1).
```
Comment (RedbackThomson) [inline docs/contents/dev-docs/release.md:14]: From Slack: "all you would need to do is create a new release on the Github repo" - Create a release (with what as the message?) and that should subsequently create a tag.

Comment (RedbackThomson) [reply docs/contents/dev-docs/release.md:14]: I would also add here some documentation linking to SemVer, so teams know when to push a patch vs. a minor release.

**Review by RedbackThomson — CHANGES_REQUESTED** [review]

File: docs/contents/dev-docs/release.md (hunk 2 above)
Comment (vijtrip2) [reply docs/contents/dev-docs/release.md:14]: For 1) I pointed that steps for cutting a release are mentioned in the document below.
For 2) That information is already present in another doc. I have linked it.

Good suggestions. Thanks.

**Review by A-Hilaly — APPROVED** [review]
👍

**Review by RedbackThomson — APPROVED** [review]

**Pull request merged**